### PR TITLE
[RFR] Add filters buttons & display filters

### PR DESCRIPTION
**Pull request by manuquentin**
- [x] Add filters button
- [x] Display filter
- [x] Update route
- [x] Update results
- [x] Add tests

Screens:
filters

### CODE REVIEW

File: app/Component/Button/FilterButton.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        let displayedFilters = this.state.data.get('filters');
+        const displayFilter = this.displayFilter.bind(this);
+
+        if (!filters || !filters.length) {
+            return null;
+        }
+
+        filters = new Set(filters);
+        displayedFilters = new Set(displayedFilters);
+        filters = new Set([...filters].filter(x => !displayedFilters.has(x)));
```
Comment (jeromemacias) [inline app/Component/Button/FilterButton.js]: Can't you do that in the store ?

File: app/Component/Datagrid/Filters.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    addFilter(filter) {
+        ApplicationActions.displayFilter(filter);
+    }
+
+    removeFilter(filter) {
+        const updateField = this.updateField.bind(this);
+
+        return () => {
+            ApplicationActions.removeFilter(filter);
+            updateField(filter.name(), null);
```
Comment (jeromemacias) [inline app/Component/Datagrid/Filters.js]: You can write `this.updateField(filter.name(), null)`.

File: app/Component/Field/InputField.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 
 class InputField extends React.Component {
     onChange(e) {
+        console.log('changed');
```
Comment (jeromemacias) [inline app/Component/Field/InputField.js]: To remove.

File: app/Component/Datagrid/Filters.js (hunk 2 above)
Comment (manuquentin) [inline app/Component/Datagrid/Filters.js]: Nope, it's an anonymous function so the this doesn't refers to the component.

File: app/Component/Button/FilterButton.js (hunk 1 above)
Comment (manuquentin) [inline app/Component/Button/FilterButton.js]: `filters` comes from props, the stores is not aware of them

File: app/Component/Datagrid/Filters.js (hunk 2 above)
Comment (jeromemacias) [inline app/Component/Datagrid/Filters.js]: From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions:
```
An arrow function expression (also known as fat arrow function) has a shorter syntax compared to function expressions and lexically binds the this value. Arrow functions are always anonymous.
```

Comment (manuquentin) [inline app/Component/Datagrid/Filters.js]: OK, my bad

**Pull request merged**